File: c/easy/transformed_array.c
#include <stdlib.h>

int* constructTransformedArray(int* nums, int n) {
    int* result = (int*)malloc(n * sizeof(int));
    if (!result) return NULL;

    for (int i = 0; i < n; i++) {
        int j = (i + nums[i]) % n;
        result[i] = nums[j];
    }

    return result;
}

File: rust/easy/transformed_array.rs
fn construct_transformed_array(nums: &Vec<i32>) -> Vec<i32> {
    let n = nums.len();
    let mut result = vec![0; n];

    for i in 0..n {
        let j = ((i as i32 + nums[i]) % n as i32) as usize;
        result[i] = nums[j];
    }

    result
}
